aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterQueryTest.java

```diff
@@ -146,6 +147,46 @@ public void querySavedDataWithForeignKey() throws DataStoreException {
         assertTrue(blogs.contains(blog));
     }
 
+    /**
+     * Test that querying the saved item with a foreign key
+     * also populates that instance variable with object.
+     * @throws DataStoreException On unexpected failure manipulating items in/out of DataStore
+     */
+    @Test
+    public void querySavedDataWithMultiLevelJoins() throws DataStoreException {
+        final BlogOwner blogOwner = BlogOwner.builder()
+            .name("Alan Turing")
+            .build();
+
+        final Blog blog = Blog.builder()
+            .name("Alan's Software Blog")
+            .owner(blogOwner)
+            .build();
+
+        final Post post = Post.builder()
+            .title("Alan's first post")
+            .status(PostStatus.ACTIVE)
+            .rating(2)
+            .blog(blog)
+            .build();
+
+        final Comment comment = Comment.builder()
+            .content("Alan's first comment")
+            .post(post)
+            .build();
+
+        adapter.save(blogOwner);
+        adapter.save(blog);
+        adapter.save(post);
+        adapter.save(comment);
```

Comment:
If you like it, this should work, too:
```suggestion
        adapter.save(blogOwner, blog, post, comment);
```